### ISSUE TITLE
How to load components on demand with JSPM?

### ISSUE THREAD

**Issue by playground** [jspm/jspm-cli#1435]
This is not an issue.  Want to know if JSPM supports on demand loading...
@guybedford is there a way in JSPM that we can configure to indicate what components should be loaded on demand or via user actions?  For example, in app.js

import 'reflect-metadata';
import {provide, Component} from 'angular2/core';
import {bootstrap} from 'angular2/platform/browser';

// At the top level, set the location strategy.
import {ROUTER_PROVIDERS, LocationStrategy, HashLocationStrategy,
  RouteConfig, Router, Location,
  ROUTER_DIRECTIVES} from 'angular2/router';
// RouterOutlet is in ROUTER_DIRECTIVES
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/zip';

import '../../css/core.css!';
import '../../css/app.css!';
import { _settings } from '../../settings'
import {Sidebar} from '../sidebar/sidebar';
import {NavBar} from '../nav-bar/nav-bar';
import {Welcome} from '../welcome/welcome';
import {List} from '../list/list';
import {Blog} from '../blog/blog';
import {Header} from '../header/header';
import {Favorites} from '../favorites/favorites';
import {Map} from '../map/map'; 
import {Footer} from '../footer/footer';

Would like to only load in the components that are in view, any components that are below the fold should be deferred until user scroll.

**Comment by theefer** [comment]
`import`s are static dependencies; what you want is dynamic dependencies, which you can achieve using calls to `System.import` (`System.import('the-module').then(theModule)`) whenever you need them.

**Comment by twoixter** [comment]
@theefer Is there a way for a static dependency to wait on a dynamic dependency? I see the `System.import` is fully asynchronous. Take for example a i18n library that depends dynamically on the language.

```language javascript
// The main import
import __ from i18n;
```

```language javascript
// The i18n
System.import("/locale/" + the_lang()).then((trans) => {
    export default trans;  // Can't be done!
});
```

**Comment by theefer** [comment]
I don't think it can be done the way you're describing it in ES6 due to the static nature of the import/export semantics.

It may be possible in CommonJS (or possibly AMD), though I see [you have tried](https://github.com/jspm/jspm-cli/issues/1439). Might be worth closing this issue as "can't be done (in ES6)" and moving further discussions to the other issue.
